cloud/new_ec2_client.go

```diff
@@ -94,6 +102,12 @@ func (c *awsClientImpl) RunInstances(input *ec2.RunInstancesInput) (*ec2.Reserva
 		func() (bool, error) {
 			output, err = c.EC2.RunInstances(input)
 			if err != nil {
+				if ec2err, ok := err.(awserr.Error); ok {
```

Comment:
should these errors be checking ec2err != nil rather than ok?

Reply:
The error is not nil from the line above. ok verifies that the type assertion was successful. From the docs: "Returns awserr.Error for service API and SDK errors. Use runtime type assertions with awserr.Error's Code and Message methods to get detailed information about the error." (https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/)